### DISCUSSION TITLE
0.9.5

### DISCUSSION THREAD

**Discussion by kamiyaa** [kamiyaa/joshuto#355] [Announcements]
Thank you everyone who have contributed and supported to this project.
It has been over a year since I made a new release.
In that time, the amount of users using Joshuto has grown a lot and a lot of new features has been added.
I hope to continue improving on this project and working together with everyone!

## Changes
- `delete_files` now has the ability to be queued up (just like cut/copy operations)
- `--file-chooser` option for using joshuto as a file picker
- add `symlink_files` command
  - works like cut/copy but instead of pasting the file, paste an absolute or relative symlink to the targets
- File sorting now works per tab
  - Thanks to @DLFW 
- Visual mode support
  - Thanks to @DLFW 
- `numbered_command` is now a proper command that can be invoked
- `tab_switch_index` open tabs to populate given index
- Show how many files are selected in the footer
- Mimetype support! (require `file` command installed to work)
- Add Ranger-like mouse controls
  - Thanks to @DLFW 
- `filter` command to filter directory listing
  - Thanks to @krivahtoo 
- Add linemode
  - Thanks to @DLFW 
- Support multiple zoxide arguments
  - Thanks to @krivahtoo 
- `rename_keep_ext`: rename with the file name removed but keeps extension
  - Thanks to @fritzrehde 
- Bookmark support
- `--permanently` option for deleting files
  - Thanks to @krivahtoo 
- Screenshot compression
  - Thanks to @indigolain 
- Remove deleted file entries from history
  - Thanks to @sergey-suslov 
- New tab can now open in specific directories
  - Thanks to @DLFW 
- Ability to quit to directory via command line argument on startup
  - Thanks to @pantosaur 
- Migration to `ratatui`
  - Thanks to @Ragnyll and @MathieuHerrmann 
- Nix flake support
  - Thanks to @Ruixi-rebirth 
- Add support for customizing file icons
  - Thanks to @krivahtoo 
- Migration to `termion` v2
  - Thanks to @asim215 
- Migrate to shellexpand v3 and rustyline v12
- Avoid changing constraints during preview loading
  - Thanks to @Handfish 

- Documentation fixes and updates
  - Thanks to @aouerf , @DLFW , @yanshay , @danielronalds , @a-kenji , @eltociear , @hezhizhen , @imbolc , @donovanglover 

## Fixes
- Configs are loaded at the right time now (on startup instead of when used)
- Fix preview width is 0
- Fix HSplit view crashing
  - Thanks to @Vinegret43 
- Fix `subdir_fzf` stuck on large directories
  - Thanks to @Jacky-Lzx 
- Fix incorrect `sort_method` in default config
  - Thanks to @rqdmap 

<hr /><em>This discussion was created from the release <a href='https://github.com/kamiyaa/joshuto/releases/tag/'>0.9.5</a>.</em>

**Comment by andoArbST** [comment]
This utility is great! Thanks for doing/supporting it

It seems the OSX binary wasn't included in the release 😢 

A workflow is still hanging, waiting for a worker https://github.com/kamiyaa/joshuto/actions/runs/5458253034/jobs/9933143697

**Reply by kamiyaa** [reply]
:cry: not sure what happened

**Reply by andoArbST** [reply]
Maybe just a GitHub hiccup
I can wait for the next release or build/install with cargo

Thanks!
